File: pow.rb
@cache = {}
def pow a, b
  return 1 if b.zero?
  return a if b.eql? 1
  if @cache[[a, b]]
    @cache[[a, b]]
  else
    @cache[[a, b]] = a * pow(a, b - 1)
  end
end

File: pow.js
var cache = {};
var pow = function(a, b){
  if(b == 0) return 1;
  if(b == 1) return a;

  if (cache[[a, b]]) {
    return cache[[a, b]]
  } else {
    return cache[[a, b]] = a * pow(a, b - 1);
  }
};
